src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1368,6 +1371,24 @@ static UCS_CLASS_INIT_FUNC(uct_dc_mlx5_iface_t, uct_md_h tl_md, uct_worker_h wor
     }
     ucs_assert(self->tx.num_dci_pools <= UCT_DC_MLX5_IFACE_MAX_DCI_POOLS);
 
+    /* allocate dcis arrays and stack */
+    self->tx.dcis = ucs_calloc(UCT_DC_MLX5_IFACE_MAX_DCIS(self->tx.ndci,
+                               self->tx.num_dci_pools), sizeof(*self->tx.dcis),
+                               "dcis");
+    if (self->tx.dcis == NULL) {
+        return UCS_ERR_NO_MEMORY;
```

Comment:
change labels as follows:
1. rename 'err:' to 'err_free_pools:'
2. add new label 'err:' that will only 'return status'
3. here, do 'status = UCS_ERR_NO_MEMORY' and 'goto err'

Reply:
done